src/Socket.cpp

```diff
@@ -114,6 +126,36 @@ ResolvedAddr::~ResolvedAddr()
     }
 }
 
+PeerAddr::PeerAddr(bool isIpv6, unsigned short port)
+{
+    if(isIpv6)
+    {
+        throw std::logic_error("Broadcast over IPv6 not supported!");

```

Comment:
I know. However I did not want to implement and test IPv6 multicast usage and the whole discovery approach yet.
For IPv4 it is working and IPv4 is currently still widely available especially in LAN environments. So I'd leave that feature "add IPv6 discovery" for the future.